### PR TITLE
Add new properties to gitlab.project

### PR DESCRIPTION
**Pull request by tas50**
Add a pile of new properties we need to improve the security policy + asset overview data.

```
gitlab.group.projects.first: {
  webURL: "https://gitlab.com/lunaelectric/rover"
  issuesEnabled: true
  wikiEnabled: true
  requirementsEnabled: false
  path: "rover"
  createdAt: 2023-09-20 15:56:18.042 -0700 PDT
  containerRegistryEnabled: true
  name: "Rover"
  visibility: "private"
  id: 50575105
  defaultBranch: "main"
  description: ""
  archived: false
  serviceDeskEnabled: true
  emailsDisabled: false
  mirror: false
  allowMergeOnSkippedPipeline: false
  autoDevopsEnabled: false
  onlyAllowMergeIfAllDiscussionsAreResolved: false
  onlyAllowMergeIfPipelineSucceeds: false
  fullName: "Lunalectric / Rover"
  packagesEnabled: true
  mergeRequestsEnabled: true
  snippetsEnabled: true
}
```

Highlights:

- Update the defaults to use the new fullName property which includes the namespace (group in our context)
- Include super handy things for asset inventory like the URL, create date, and status of features, if its a mirror/archived, and the default branch name.
- Include some rules for merges

### CODE REVIEW

**Review by misterpantz — APPROVED** [review]

**Comment by tas50** [comment]
We need https://github.com/mondoohq/cnquery/issues/1789 fixed before I can test this one out

File: providers/gitlab/resources/gitlab.lr [hunk 1]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
   path string
+  // Create date of the project
+  createdAt time  
   // Project description
   description string
+  // default git branch
+  defaultBranch string
   // The project's visibility level. Can be private, internal, or public.
   visibility string
+  // Is the project archived
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
  // Is the project archived?
```

File: providers/gitlab/resources/gitlab.lr [hunk 2]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  archived bool
+  // Is the project a mirror
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
  // Is the project a mirror?
```

File: providers/gitlab/resources/gitlab.lr [hunk 3]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  webURL string
+  // Disable project email notifications
+  emailsDisabled bool
+  // Allow merging merge requests when a pipeline is skipped
+  allowMergeOnSkippedPipeline bool
+  // Only allow merging merge requests if the pipelines succeed
+  onlyAllowMergeIfPipelineSucceeds bool
+  // Allow merging merge requests if all discussions are resolved
+  onlyAllowMergeIfAllDiscussionsAreResolved bool
+  // Is the issues feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
  // Is the issues feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 4]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  issuesEnabled bool
+	// Is the merge request feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
	// Is the merge request feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 5]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  mergeRequestsEnabled bool
+	// Is the wiki feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
	// Is the wiki feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 6]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  wikiEnabled bool
+	// Is the snippets feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
	// Is the snippets feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 7]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  snippetsEnabled bool
+	// Is the container registry feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
	// Is the container registry feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 8]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  containerRegistryEnabled bool
+	// Is the Service Desk feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
	// Is the Service Desk feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 9]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  serviceDeskEnabled bool
+	// Is the packages feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
	// Is the packages feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 10]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  packagesEnabled bool
+	// Is the Auto DevOps feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
	// Is the Auto DevOps feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 11]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
...
+  autoDevopsEnabled bool
+	// Is the requirements feature enabled
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
	// Is the requirements feature enabled?
```

File: providers/gitlab/resources/gitlab.lr [hunk 12]
```diff
@@ -31,15 +31,53 @@ gitlab.group @defaults("name") {
 }
 
 // GitLab Project
-gitlab.project @defaults("name visibility") {
+gitlab.project @defaults("fullName visibility webURL") {
   // Project ID
   id int
   // Project name
   name string
+  // The full name of the project including the namespace
```
Comment (misterpantz) [inline providers/gitlab/resources/gitlab.lr]:
```suggestion
  // The full name of the project, including the namespace
```

**Review by misterpantz — APPROVED** [review]

**Review by arlimus — COMMENTED** [review]
Awesome, more data!

**Pull request merged**